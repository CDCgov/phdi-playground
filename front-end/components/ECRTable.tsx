import {Table} from '@trussworks/react-uswds'
import _ from 'lodash';

export default function ECRTable({ecrData}) {
	const options = ['patient_id', 'first_name', 'last_name', 'gender', 'birth_date']

	console.log(ecrData);

	const getTableBody = (data: any) => {
		return (
			<tbody>
			{options.map(function (option) {
				return (
					<tr key={option}>
						<th scope="row">{_.startCase(option)}</th>
						<td>{data[option]}</td>
					</tr>
				);
			})}

			</tbody>
		)
	}

	return (
		<div className='margin-3'>
			{/*<h1>eCR Viewer</h1>*/}
			<div>
				<Table
					bordered
					// caption="This table uses the fullWidth prop to increase to 100% width"
					fullWidth>
					<thead>
					<tr>
						<th scope="col">Field Name</th>
						<th scope="col">Field Value</th>
					</tr>
					</thead>
					{getTableBody(ecrData.processed_values.parsed_values)}
				</Table>
			</div>
		</div>
	)
}