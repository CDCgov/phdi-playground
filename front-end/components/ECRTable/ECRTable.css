.usa-table {
    max-width: 512px;
}