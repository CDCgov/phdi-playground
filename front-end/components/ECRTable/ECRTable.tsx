import {
	Table
} from '@trussworks/react-uswds'
import _ from 'lodash';
import { v4 } from 'uuid';


export default function ECRTable({ ecrData }: any) {
	const tableData = ecrData?.processed_values?.parsed_values
	// remove null fields
	Object.keys(tableData || {}).forEach((k) => tableData[k] == null && delete tableData[k])
	const options = Object.keys(tableData || {});
	const arrayOptions = options.filter((option) => Array.isArray(tableData[option]))
	// remove array fields from options
	arrayOptions.forEach(arrayOption => {options.splice(options.indexOf(arrayOption), 1)})
	const formatHeader = (innerOption: string) => _.startCase(innerOption)
		.replace(/rr /i, "RR ")
		.replace(/eicr /i, "eICR ")
		.replace(/ecr /i, "eCR ");

	const getArrayTableHeader = (data: [{[key: string]: any}]) => {
		const innerOptions = data.length > 0 && Object.keys(data[0] || {}) || [];

		return (
			<thead>
				<tr>
					{innerOptions.map(innerOption => <th scope="col" key={v4()}>{formatHeader(innerOption)}</th>)}
				</tr>
			</thead>
		)
	}

	const getArrayTableBody = (data: [{[key: string]: any}]) => {
		const innerOptions = data.length > 0 && Object.keys(data[0] || {}) || [];

		return (
			<tbody>
			{data.map(entry => (
				<tr key={v4()}>
					{innerOptions.map(option => {
						return (<th scope="row" key={v4()}>{entry[option]}</th>)
					})}
				</tr>
			))}

			</tbody>
		)
	}

	const getTableBody = (data: any) => {
		Object.keys(data || {}).forEach((k) => data[k] == null && delete data[k])
		const innerOptions = Object.keys(data || {});
		const innerArrayOptions = innerOptions.filter((option) => Array.isArray(data[option]))
		innerArrayOptions.forEach(arrayOption => {innerOptions.splice(innerOptions.indexOf(arrayOption), 1)})
		return (
			<tbody>
			{data && innerOptions.map(function (option) {
				return (
					<tr key={option}>
						<th scope="row">{formatHeader(option)}</th>
						<td>{data[option]}</td>
					</tr>
				);
			})}

			</tbody>
		)
	}

	return (
		<div className='margin-top-3'>
			<h1>eCR Viewer</h1>
			<div>
				<Table
					bordered
					caption="This table uses the fullWidth prop to increase to 100% width"
					fullWidth>
					<thead>
					<tr>
						<th scope="col">Field Name</th>
						<th scope="col">Field Value</th>
					</tr>
					</thead>
					{getTableBody(tableData)}
				</Table>
			</div>
			{arrayOptions.map(option => {
				return (
					<div key={option}>
						<h2>{formatHeader(option)}</h2>
						<div>
							<Table
								bordered
								scrollable
								fullWidth
								className="no-wrap-table"
							>
								{getArrayTableHeader(tableData[option])}
								{getArrayTableBody(tableData[option])}
							</Table>
						</div>
					</div>)
			})}
		</div>
	)
}