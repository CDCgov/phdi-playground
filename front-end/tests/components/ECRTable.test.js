import React from 'react';
import { render, screen } from '@testing-library/react';
import "@testing-library/jest-dom";
import ECRTable from '../../components/ECRTable'; // Adjust the import path as per your project structure

describe('ECRTable', () => {
    it('renders ECRTable component with provided data', () => {
        const ecrData = {
            "processed_values": {
                "message": "Parsing succeeded!",
                "parsed_values": {
                    "patient_id": "123",
                    "person_id": null,
                    "last_name": "BEAKER",
                    "first_name": "CATEST",
                    "birth_date": "2016-01-27",
                    "gender": "female"
                }
            }
        };

        render(<ECRTable ecrData={ecrData} />);

        // You can add more specific assertions here
        // For example, checking if the table headers and data are rendered
        expect(screen.getByText('eCR Viewer')).toBeInTheDocument();
        expect(screen.getByText('Field Name')).toBeInTheDocument();
        expect(screen.getByText('Field Value')).toBeInTheDocument();
        expect(screen.getByText('Patient Id')).toBeInTheDocument();
        expect(screen.getByText('123')).toBeInTheDocument();
        // Add similar assertions for other data fields
    });
});
