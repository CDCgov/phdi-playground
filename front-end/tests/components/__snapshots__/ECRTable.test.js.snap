// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ECRTable renders ECRTable component with ALL provided data 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="margin-top-3"
      >
        <h1>
          eCR Viewer
        </h1>
        <div>
          <table
            class="usa-table width-full"
            data-testid="table"
          >
            <caption>
              This table uses the fullWidth prop to increase to 100% width
            </caption>
            <thead>
              <tr>
                <th
                  scope="col"
                >
                  Field Name
                </th>
                <th
                  scope="col"
                >
                  Field Value
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <th
                  scope="row"
                >
                  Patient Id
                </th>
                <td>
                  70c659b7-675f-4d51-9a75-2111a88dbcc8
                </td>
              </tr>
              <tr>
                <th
                  scope="row"
                >
                  Last Name
                </th>
                <td>
                  ZZTESTTWELVE
                </td>
              </tr>
              <tr>
                <th
                  scope="row"
                >
                  First Name
                </th>
                <td>
                  CEDARSECR
                </td>
              </tr>
              <tr>
                <th
                  scope="row"
                >
                  Birth Date
                </th>
                <td>
                  1990-12-31
                </td>
              </tr>
              <tr>
                <th
                  scope="row"
                >
                  Gender
                </th>
                <td>
                  female
                </td>
              </tr>
              <tr>
                <th
                  scope="row"
                >
                  Race
                </th>
                <td>
                  Native Hawaiian or Other Pacific Islander,Asian
                </td>
              </tr>
              <tr>
                <th
                  scope="row"
                >
                  Ethnicity
                </th>
                <td>
                  Hispanic
                </td>
              </tr>
              <tr>
                <th
                  scope="row"
                >
                  Street Address 1
                </th>
                <td>
                  1123 MALTMAN STREET
                </td>
              </tr>
              <tr>
                <th
                  scope="row"
                >
                  State
                </th>
                <td>
                  CA
                </td>
              </tr>
              <tr>
                <th
                  scope="row"
                >
                  Zip
                </th>
                <td>
                  90026
                </td>
              </tr>
              <tr>
                <th
                  scope="row"
                >
                  Rr Id
                </th>
                <td>
                  3f854741-c9f7-462a-8ee5-590a72b9af33
                </td>
              </tr>
              <tr>
                <th
                  scope="row"
                >
                  Status
                </th>
                <td>
                  RRVS19
                </td>
              </tr>
              <tr>
                <th
                  scope="row"
                >
                  Conditions
                </th>
                <td>
                  Disease caused by severe acute respiratory syndrome coronavirus 2 (disorder),Disease caused by severe acute respiratory syndrome coronavirus 2 (disorder)
                </td>
              </tr>
              <tr>
                <th
                  scope="row"
                >
                  Eicr Set Id
                </th>
                <td>
                  69bbc532-0a17-11ed-94c7-005056a7fb21
                </td>
              </tr>
              <tr>
                <th
                  scope="row"
                >
                  Eicr Id
                </th>
                <td>
                  1.2.840.114350.1.13.202.3.7.8.688883.266
                </td>
              </tr>
              <tr>
                <th
                  scope="row"
                >
                  Eicr Version Number
                </th>
                <td>
                  16
                </td>
              </tr>
              <tr>
                <th
                  scope="row"
                >
                  Replaced Eicr Id
                </th>
                <td>
                  Composition/1.2.840.114350.1.13.202.3.7.8.688883.265
                </td>
              </tr>
              <tr>
                <th
                  scope="row"
                >
                  Replaced Eicr Version
                </th>
                <td>
                  15
                </td>
              </tr>
              <tr>
                <th
                  scope="row"
                >
                  Authoring Datetime
                </th>
                <td>
                  2022-07-25T16:20:31-07:00
                </td>
              </tr>
              <tr>
                <th
                  scope="row"
                >
                  Facility Id Number
                </th>
                <td>
                  1.2.840.114350.1.13.202.3.7.2.686980
                </td>
              </tr>
              <tr>
                <th
                  scope="row"
                >
                  Facility Name
                </th>
                <td>
                  CAR TOW635E CSMG
                </td>
              </tr>
              <tr>
                <th
                  scope="row"
                >
                  Facility Type
                </th>
                <td>
                  Healthcare Facility
                </td>
              </tr>
              <tr>
                <th
                  scope="row"
                >
                  Encounter Start Date
                </th>
                <td>
                  2022-01-21
                </td>
              </tr>
              <tr>
                <th
                  scope="row"
                >
                  Encounter End Date
                </th>
                <td>
                  2022-01-21
                </td>
              </tr>
              <tr>
                <th
                  scope="row"
                >
                  Reason For Visit
                </th>
                <td>
                  Not on file
                </td>
              </tr>
              <tr>
                <th
                  scope="row"
                >
                  Preferred Language
                </th>
                <td>
                  English
                </td>
              </tr>
            </tbody>
          </table>
        </div>
        <div>
          <h2>
            Active Problems
          </h2>
          <div>
            <div
              class="usa-table-container--scrollable"
              data-testid="scrollable-table-container"
              tabindex="0"
            >
              <table
                class="usa-table width-full no-wrap-table"
                data-testid="table"
              >
                <thead>
                  <tr>
                    <th
                      scope="col"
                    >
                      Problem
                    </th>
                    <th
                      scope="col"
                    >
                      Problem Date
                    </th>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <th
                      scope="row"
                    >
                      Mild to moderate pre-eclampsia
                    </th>
                    <th
                      scope="row"
                    >
                      2022-07-21
                    </th>
                  </tr>
                  <tr>
                    <th
                      scope="row"
                    >
                      Depression, recurrent
                    </th>
                    <th
                      scope="row"
                    >
                      2022-07-21
                    </th>
                  </tr>
                  <tr>
                    <th
                      scope="row"
                    >
                      Pain, upper back
                    </th>
                    <th
                      scope="row"
                    >
                      2022-07-22
                    </th>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
        <div>
          <h2>
            Labs
          </h2>
          <div>
            <div
              class="usa-table-container--scrollable"
              data-testid="scrollable-table-container"
              tabindex="0"
            >
              <table
                class="usa-table width-full no-wrap-table"
                data-testid="table"
              >
                <thead>
                  <tr>
                    <th
                      scope="col"
                    >
                      Test Type
                    </th>
                    <th
                      scope="col"
                    >
                      Test Type Code
                    </th>
                    <th
                      scope="col"
                    >
                      Test Type System
                    </th>
                    <th
                      scope="col"
                    >
                      Test Result Qualitative
                    </th>
                    <th
                      scope="col"
                    >
                      Test Result Quantitative
                    </th>
                    <th
                      scope="col"
                    >
                      Test Result Units
                    </th>
                    <th
                      scope="col"
                    >
                      Test Result Code
                    </th>
                    <th
                      scope="col"
                    >
                      Test Result Code Display
                    </th>
                    <th
                      scope="col"
                    >
                      Test Result Code System
                    </th>
                    <th
                      scope="col"
                    >
                      Test Result Interp
                    </th>
                    <th
                      scope="col"
                    >
                      Test Result Interp Code
                    </th>
                    <th
                      scope="col"
                    >
                      Test Result Interp System
                    </th>
                    <th
                      scope="col"
                    >
                      Test Result Ref Range Low
                    </th>
                    <th
                      scope="col"
                    >
                      Test Result Ref Range Low Units
                    </th>
                    <th
                      scope="col"
                    >
                      Test Result Ref Range High
                    </th>
                    <th
                      scope="col"
                    >
                      Test Result Ref Range High Units
                    </th>
                    <th
                      scope="col"
                    >
                      Specimen Type
                    </th>
                    <th
                      scope="col"
                    >
                      Performing Lab
                    </th>
                    <th
                      scope="col"
                    >
                      Specimen Collection Date
                    </th>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <th
                      scope="row"
                    >
                      SARS-CoV-2, RNA
                    </th>
                    <th
                      scope="row"
                    >
                      94500-6
                    </th>
                    <th
                      scope="row"
                    >
                      http://loinc.org
                    </th>
                    <th
                      scope="row"
                    >
                      Detected
                    </th>
                    <th
                      scope="row"
                    />
                    <th
                      scope="row"
                    />
                    <th
                      scope="row"
                    />
                    <th
                      scope="row"
                    />
                    <th
                      scope="row"
                    />
                    <th
                      scope="row"
                    >
                      Abnormal
                    </th>
                    <th
                      scope="row"
                    >
                      A
                    </th>
                    <th
                      scope="row"
                    >
                      http://terminology.hl7.org/CodeSystem/v3-ObservationInterpretation
                    </th>
                    <th
                      scope="row"
                    />
                    <th
                      scope="row"
                    />
                    <th
                      scope="row"
                    />
                    <th
                      scope="row"
                    />
                    <th
                      scope="row"
                    />
                    <th
                      scope="row"
                    >
                      LABCORP
                    </th>
                    <th
                      scope="row"
                    >
                      2022-01-21T08:00:00Z
                    </th>
                  </tr>
                  <tr>
                    <th
                      scope="row"
                    >
                      Lab Interpretation
                    </th>
                    <th
                      scope="row"
                    >
                      56850-1
                    </th>
                    <th
                      scope="row"
                    >
                      http://loinc.org
                    </th>
                    <th
                      scope="row"
                    >
                      Abnormal
                    </th>
                    <th
                      scope="row"
                    />
                    <th
                      scope="row"
                    />
                    <th
                      scope="row"
                    />
                    <th
                      scope="row"
                    />
                    <th
                      scope="row"
                    />
                    <th
                      scope="row"
                    />
                    <th
                      scope="row"
                    />
                    <th
                      scope="row"
                    />
                    <th
                      scope="row"
                    />
                    <th
                      scope="row"
                    />
                    <th
                      scope="row"
                    />
                    <th
                      scope="row"
                    />
                    <th
                      scope="row"
                    />
                    <th
                      scope="row"
                    />
                    <th
                      scope="row"
                    >
                      2022-01-21T08:00:00Z
                    </th>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="margin-top-3"
    >
      <h1>
        eCR Viewer
      </h1>
      <div>
        <table
          class="usa-table width-full"
          data-testid="table"
        >
          <caption>
            This table uses the fullWidth prop to increase to 100% width
          </caption>
          <thead>
            <tr>
              <th
                scope="col"
              >
                Field Name
              </th>
              <th
                scope="col"
              >
                Field Value
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <th
                scope="row"
              >
                Patient Id
              </th>
              <td>
                70c659b7-675f-4d51-9a75-2111a88dbcc8
              </td>
            </tr>
            <tr>
              <th
                scope="row"
              >
                Last Name
              </th>
              <td>
                ZZTESTTWELVE
              </td>
            </tr>
            <tr>
              <th
                scope="row"
              >
                First Name
              </th>
              <td>
                CEDARSECR
              </td>
            </tr>
            <tr>
              <th
                scope="row"
              >
                Birth Date
              </th>
              <td>
                1990-12-31
              </td>
            </tr>
            <tr>
              <th
                scope="row"
              >
                Gender
              </th>
              <td>
                female
              </td>
            </tr>
            <tr>
              <th
                scope="row"
              >
                Race
              </th>
              <td>
                Native Hawaiian or Other Pacific Islander,Asian
              </td>
            </tr>
            <tr>
              <th
                scope="row"
              >
                Ethnicity
              </th>
              <td>
                Hispanic
              </td>
            </tr>
            <tr>
              <th
                scope="row"
              >
                Street Address 1
              </th>
              <td>
                1123 MALTMAN STREET
              </td>
            </tr>
            <tr>
              <th
                scope="row"
              >
                State
              </th>
              <td>
                CA
              </td>
            </tr>
            <tr>
              <th
                scope="row"
              >
                Zip
              </th>
              <td>
                90026
              </td>
            </tr>
            <tr>
              <th
                scope="row"
              >
                Rr Id
              </th>
              <td>
                3f854741-c9f7-462a-8ee5-590a72b9af33
              </td>
            </tr>
            <tr>
              <th
                scope="row"
              >
                Status
              </th>
              <td>
                RRVS19
              </td>
            </tr>
            <tr>
              <th
                scope="row"
              >
                Conditions
              </th>
              <td>
                Disease caused by severe acute respiratory syndrome coronavirus 2 (disorder),Disease caused by severe acute respiratory syndrome coronavirus 2 (disorder)
              </td>
            </tr>
            <tr>
              <th
                scope="row"
              >
                Eicr Set Id
              </th>
              <td>
                69bbc532-0a17-11ed-94c7-005056a7fb21
              </td>
            </tr>
            <tr>
              <th
                scope="row"
              >
                Eicr Id
              </th>
              <td>
                1.2.840.114350.1.13.202.3.7.8.688883.266
              </td>
            </tr>
            <tr>
              <th
                scope="row"
              >
                Eicr Version Number
              </th>
              <td>
                16
              </td>
            </tr>
            <tr>
              <th
                scope="row"
              >
                Replaced Eicr Id
              </th>
              <td>
                Composition/1.2.840.114350.1.13.202.3.7.8.688883.265
              </td>
            </tr>
            <tr>
              <th
                scope="row"
              >
                Replaced Eicr Version
              </th>
              <td>
                15
              </td>
            </tr>
            <tr>
              <th
                scope="row"
              >
                Authoring Datetime
              </th>
              <td>
                2022-07-25T16:20:31-07:00
              </td>
            </tr>
            <tr>
              <th
                scope="row"
              >
                Facility Id Number
              </th>
              <td>
                1.2.840.114350.1.13.202.3.7.2.686980
              </td>
            </tr>
            <tr>
              <th
                scope="row"
              >
                Facility Name
              </th>
              <td>
                CAR TOW635E CSMG
              </td>
            </tr>
            <tr>
              <th
                scope="row"
              >
                Facility Type
              </th>
              <td>
                Healthcare Facility
              </td>
            </tr>
            <tr>
              <th
                scope="row"
              >
                Encounter Start Date
              </th>
              <td>
                2022-01-21
              </td>
            </tr>
            <tr>
              <th
                scope="row"
              >
                Encounter End Date
              </th>
              <td>
                2022-01-21
              </td>
            </tr>
            <tr>
              <th
                scope="row"
              >
                Reason For Visit
              </th>
              <td>
                Not on file
              </td>
            </tr>
            <tr>
              <th
                scope="row"
              >
                Preferred Language
              </th>
              <td>
                English
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div>
        <h2>
          Active Problems
        </h2>
        <div>
          <div
            class="usa-table-container--scrollable"
            data-testid="scrollable-table-container"
            tabindex="0"
          >
            <table
              class="usa-table width-full no-wrap-table"
              data-testid="table"
            >
              <thead>
                <tr>
                  <th
                    scope="col"
                  >
                    Problem
                  </th>
                  <th
                    scope="col"
                  >
                    Problem Date
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <th
                    scope="row"
                  >
                    Mild to moderate pre-eclampsia
                  </th>
                  <th
                    scope="row"
                  >
                    2022-07-21
                  </th>
                </tr>
                <tr>
                  <th
                    scope="row"
                  >
                    Depression, recurrent
                  </th>
                  <th
                    scope="row"
                  >
                    2022-07-21
                  </th>
                </tr>
                <tr>
                  <th
                    scope="row"
                  >
                    Pain, upper back
                  </th>
                  <th
                    scope="row"
                  >
                    2022-07-22
                  </th>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
      <div>
        <h2>
          Labs
        </h2>
        <div>
          <div
            class="usa-table-container--scrollable"
            data-testid="scrollable-table-container"
            tabindex="0"
          >
            <table
              class="usa-table width-full no-wrap-table"
              data-testid="table"
            >
              <thead>
                <tr>
                  <th
                    scope="col"
                  >
                    Test Type
                  </th>
                  <th
                    scope="col"
                  >
                    Test Type Code
                  </th>
                  <th
                    scope="col"
                  >
                    Test Type System
                  </th>
                  <th
                    scope="col"
                  >
                    Test Result Qualitative
                  </th>
                  <th
                    scope="col"
                  >
                    Test Result Quantitative
                  </th>
                  <th
                    scope="col"
                  >
                    Test Result Units
                  </th>
                  <th
                    scope="col"
                  >
                    Test Result Code
                  </th>
                  <th
                    scope="col"
                  >
                    Test Result Code Display
                  </th>
                  <th
                    scope="col"
                  >
                    Test Result Code System
                  </th>
                  <th
                    scope="col"
                  >
                    Test Result Interp
                  </th>
                  <th
                    scope="col"
                  >
                    Test Result Interp Code
                  </th>
                  <th
                    scope="col"
                  >
                    Test Result Interp System
                  </th>
                  <th
                    scope="col"
                  >
                    Test Result Ref Range Low
                  </th>
                  <th
                    scope="col"
                  >
                    Test Result Ref Range Low Units
                  </th>
                  <th
                    scope="col"
                  >
                    Test Result Ref Range High
                  </th>
                  <th
                    scope="col"
                  >
                    Test Result Ref Range High Units
                  </th>
                  <th
                    scope="col"
                  >
                    Specimen Type
                  </th>
                  <th
                    scope="col"
                  >
                    Performing Lab
                  </th>
                  <th
                    scope="col"
                  >
                    Specimen Collection Date
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <th
                    scope="row"
                  >
                    SARS-CoV-2, RNA
                  </th>
                  <th
                    scope="row"
                  >
                    94500-6
                  </th>
                  <th
                    scope="row"
                  >
                    http://loinc.org
                  </th>
                  <th
                    scope="row"
                  >
                    Detected
                  </th>
                  <th
                    scope="row"
                  />
                  <th
                    scope="row"
                  />
                  <th
                    scope="row"
                  />
                  <th
                    scope="row"
                  />
                  <th
                    scope="row"
                  />
                  <th
                    scope="row"
                  >
                    Abnormal
                  </th>
                  <th
                    scope="row"
                  >
                    A
                  </th>
                  <th
                    scope="row"
                  >
                    http://terminology.hl7.org/CodeSystem/v3-ObservationInterpretation
                  </th>
                  <th
                    scope="row"
                  />
                  <th
                    scope="row"
                  />
                  <th
                    scope="row"
                  />
                  <th
                    scope="row"
                  />
                  <th
                    scope="row"
                  />
                  <th
                    scope="row"
                  >
                    LABCORP
                  </th>
                  <th
                    scope="row"
                  >
                    2022-01-21T08:00:00Z
                  </th>
                </tr>
                <tr>
                  <th
                    scope="row"
                  >
                    Lab Interpretation
                  </th>
                  <th
                    scope="row"
                  >
                    56850-1
                  </th>
                  <th
                    scope="row"
                  >
                    http://loinc.org
                  </th>
                  <th
                    scope="row"
                  >
                    Abnormal
                  </th>
                  <th
                    scope="row"
                  />
                  <th
                    scope="row"
                  />
                  <th
                    scope="row"
                  />
                  <th
                    scope="row"
                  />
                  <th
                    scope="row"
                  />
                  <th
                    scope="row"
                  />
                  <th
                    scope="row"
                  />
                  <th
                    scope="row"
                  />
                  <th
                    scope="row"
                  />
                  <th
                    scope="row"
                  />
                  <th
                    scope="row"
                  />
                  <th
                    scope="row"
                  />
                  <th
                    scope="row"
                  />
                  <th
                    scope="row"
                  />
                  <th
                    scope="row"
                  >
                    2022-01-21T08:00:00Z
                  </th>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
